### PR TITLE
node-502-skill-crud-operations

### PR DESCRIPTION
**Pull request by chadiegilnerubia**
Ticket ID:
Create API to Create / Retrieve / Update / Delete Skills
[#502](https://github.com/nerubia/kh360-nodejs/issues/502)

### CODE REVIEW

File: src/controllers/admin/skill-controller.ts [hunk 1]
```diff
@@ -22,3 +25,97 @@ export const index = async (req: Request, res: Response) => {
...
+
+/**
+ * Delete skill category
+ * @param req.params.id -The ID of the skill to be deleted
+ */
+export const destroy = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params
+    await SkillService.deletedById(parseInt(id))
+    res.json({ id, message: "Skill category successfully deleted" })
```
Comment (gborgonia-ir) [inline src/controllers/admin/skill-controller.ts:111]: update appropriate message here

File: src/controllers/admin/skill-controller.ts [hunk 2]
```diff
@@ -22,3 +25,97 @@ export const index = async (req: Request, res: Response) => {
...
+    }
+    if (error instanceof CustomError) {
+      return res.status(error.status).json({ message: error.message })
+    }
+    res.status(500).json({ message: "Something went wrong" })
+  }
+}
+
+/**
+ * Delete skill category
```
Comment (gborgonia-ir) [inline src/controllers/admin/skill-controller.ts:104]: update correct description

File: src/utils/validation/skill-schema.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { number, object, string } from "yup"
+
+export const createSkillSchema = object().shape({
+  name: string().required("Skill name is required"),
+  description: string().required("Description is required"),
```
Comment (gborgonia-ir) [inline src/utils/validation/skill-schema.ts:5]: can be pls double check on ticket if this is required

Comment (chadiegilnerubia) [reply src/utils/validation/skill-schema.ts:5]: ows mb description is not required.

File: src/controllers/admin/skill-controller.ts (hunk 2 above)
Comment (chadiegilnerubia) [reply src/controllers/admin/skill-controller.ts:104]: done :3

File: src/controllers/admin/skill-controller.ts (hunk 1 above)
Comment (chadiegilnerubia) [reply src/controllers/admin/skill-controller.ts:111]: done :3

**Review by gborgonia-ir — APPROVED** [review]

File: src/controllers/admin/skill-controller.ts [hunk 4]
```diff
@@ -22,3 +25,97 @@ export const index = async (req: Request, res: Response) => {
...
+ * @param req.body.skill_category_id
+ * @param res - return the new updated skill
+ * @returns
+ */
+export const update = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params
+    const { name, description, status, skill_category_id } = req.body
+    await createSkillSchema.validate({ name, description, status, skill_category_id })
+    const updateSkill = await SkillService.updatedById(parseInt(id), {
```
Comment (emmaacha) [inline src/controllers/admin/skill-controller.ts:85]: Kindly rename to `updateById`

File: src/controllers/admin/skill-controller.ts [hunk 5]
```diff
@@ -22,3 +25,97 @@ export const index = async (req: Request, res: Response) => {
...
+}
+
+/**
+ * Delete skill
+ * @param req.params.id -The ID of the skill to be deleted
+ */
+export const destroy = async (req: Request, res: Response) => {
+  try {
+    const { id } = req.params
+    await SkillService.deletedById(parseInt(id))
```
Comment (emmaacha) [inline src/controllers/admin/skill-controller.ts:110]: Please also rename to `deleteById`

**Review by emmaacha — CHANGES_REQUESTED** [review]

File: src/repositories/skill-repository.ts [hunk 6]
```diff
@@ -44,3 +89,11 @@ export const countByFilters = async (where: Prisma.skillsWhereInput) => {
     where,
   })
 }
+
+export const destroy = async (id: number, skill_id: number) => {
```
Comment (emmaacha) [inline src/repositories/skill-repository.ts:93]: Please rename to `deleteById`

File: src/repositories/skill-repository.ts [hunk 7]
```diff
@@ -9,12 +10,56 @@ export const getById = async (id: number) => {
...
+    data: {
+      ...data,
+      sequence_no: nextSequenceNo,
+      created_at: currentDate,
+      updated_at: currentDate,
+    },
+  })
+}
+
+export const update = async (id: number, data: Prisma.skillsUpdateInput) => {
```
Comment (emmaacha) [inline src/repositories/skill-repository.ts:39]: Please rename to updateById also

**Review by emmaacha — CHANGES_REQUESTED** [review]
Sorry kuys last najud 😅

File: src/repositories/skill-repository.ts (hunk 7 above)
Comment (chadiegilnerubia) [reply src/repositories/skill-repository.ts:39]: done :3

**Review by emmaacha — APPROVED** [review]
Cool 🚀

**Pull request merged**